### ISSUE TITLE
Error in the tpu.ipynb notebook

### ISSUE THREAD

**Issue by bjourne** [tensorflow/tensorflow#40339]
I'm trying to implement the code in this notebook https://github.com/tensorflow/docs/blob/master/site/en/guide/tpu.ipynb

The lines for updating the training loss and accuracy are incorrect:
```python
training_loss.update_state(loss * strategy.num_replicas_in_sync)
training_accuracy.update_state(labels, logits)
```
I don't understand the intent behind updating the loss with the product of the number of replicas and the batch loss but it gives the wrong result. Changing the line to
```python
training_loss.update_state(labels, logits)
```
appears to solve the bug.

I also changed the definition of `training_loss` from a `metrics.Mean` to a `metrics.SparseCategoricalCrossentropy`.

**Comment by MarkDaoust** [comment]
@rxsang can you confirm if this code is correct, and either fix or add a couple of comments explaining?

Thanks.

**Comment by rxsang** [comment]
Hi bjourne@,

Thanks for the feedbacks. Pasting all the code related to loss calculating in the current guide.

```
with tf.GradientTape() as tape:
    loss = tf.keras.losses.sparse_categorical_crossentropy(labels, logits, from_logits=True)
    loss = tf.nn.compute_average_loss(loss, global_batch_size=batch_size)
grads = tape.gradient(loss, model.trainable_variables)
optimizer.apply_gradients(list(zip(grads, model.trainable_variables)))
training_loss.update_state(loss * strategy.num_replicas_in_sync)
```

The reason is when working with DistributionStrategy, you first
1. compute the loss
2. calculate the gradients per replica
3. aggregate the gradients across all the replicas

The reason is in 3#, the aggregation is the sum of all replicas. So if your loss is a mean loss, the gradients will be `num_replicas_in_sync` larger than expected. That's why we scale the loss by `num_replicas_in_sync` in 2#. However for the reported loss, we don't want to see the scaled loss, thus we multiply `num_replicas_in_sync` back.

The way you suggest will also work, but it will compute `SparseCategoricalCrossentropy` twice. Given calculating this metric should be cheap, I can change it to this one if this is clearer.

Does that make sense to you?

**Comment by bjourne** [comment]
Thanks for your response. Two things I don't understand. Why isn't the training accuracy calculated in the same way? Shouldn't it also suffer from the same kind of scaling issues? And when I write it the way you suggest, I get losses in the range 400 to 600. So I think there must be a bug there? Does it matter that my model does not use logits?

**Comment by jvishnuvardhan** [comment]
@bjourne Can you please share a simple standalone code to reproduce your issue? Thanks!

If this was already resolved, then please close the issue. Thanks!
